src/net/sourceforge/kolmafia/textui/command/JourneyCommand.java

```diff
@@ -0,0 +1,267 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+import net.sourceforge.kolmafia.AscensionClass;
+import net.sourceforge.kolmafia.AscensionPath.Path;
+import net.sourceforge.kolmafia.KoLAdventure;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants.ZodiacZone;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.persistence.AdventureDatabase;
+import net.sourceforge.kolmafia.persistence.SkillDatabase;
+import net.sourceforge.kolmafia.session.JourneyManager;
+
+public class JourneyCommand extends AbstractCommand {
+
+  public JourneyCommand() {
+    this.usage =
+        " zones [SC | TT | PA | SA | AT | DB]| find [all | SC | TT | PA | SA | AT | DB] <skill> - Journeyman skill utility.";
+  }
+
+  @Override
+  public void run(final String cmd, final String parameters) {
+    // If you are currently a Journeyman:
+    //
+    // journey zones - show skill table for current class, indicating already learned skills
+    // journey find <skill> - show where to find the specified skill
+    //
+    // Whether or not you are a Journeyman:
+    //
+    // journey zones <class> - show skill table for specified class
+    // journey find <class> <skill> - show where to find specified skill
+    // journey find all <skill> - show where to find skill for all classes
+
+    String[] params = parameters.trim().split("\\s+");
+    if (params.length < 1) {
+      RequestLogger.printLine("Usage: journey" + this.usage);
+      return;
+    }
+
+    String subcmd = params[0];
+
+    if (subcmd.equals("zones")) {
+      zonesCommand(params);
+      return;
+    }
+
+    if (subcmd.equals("find")) {
+      findCommand(params);
+      return;
+    }
+
+    RequestLogger.printLine("Usage: journey" + this.usage);
+  }
+
+  private AscensionClass parseClass(String[] params) {
+    switch (params[1].toLowerCase().trim()) {
+      case "sc":
+        return AscensionClass.SEAL_CLUBBER;
+      case "tt":
+        return AscensionClass.TURTLE_TAMER;
+      case "pa":
```

Comment:
while this is what the usage says, I think the standard abbreviation for pastamancer is "P" or "PM".

similarly I think S or SR is more common than SA for sauceror. I would recognise SA though.

Reply:
You know, I intended to use PM. So you spotted a bug, not just an unconventional decision. :)
I have seen "S" but not "SR". I think I have seen "SA", but, yeah - "S" is more common.
I will fix both of those. both in the usage string and the code.